week2/assignment/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"sync"
+)
+
+func readFileByLine(filePath string, lines chan<-string, processingFile <-chan string, wg *sync.WaitGroup) {	
+	file, err := os.Open(filePath)
+	
+	if err != nil {
+		log.Fatal(err)
```

Comment:
it's nice that we're using `log.Fatal` here. Just be aware that if we return error instead of exit, this program would not work, because `wg.Done()` won't be called.